src/Controller/AppController.php

```diff
@@ -118,5 +118,19 @@ protected function prepareRequest() : void
             }
             unset($this->request->data['_jsonKeys']);
         }
+        // 'cart' for relations view/save - prepare api calls
+        if (!empty($data['cart'])) {
```

Comment:
`Superminor` note on naming, may be ignored. I'd prefer for clarity:
* `$data['relations`] or `'rel'` or `'related'` instead of `$data['cart']`
* same key used on `$this->request->data` instead of `'api'`